src/coreclr/src/jit/jit.h

```diff
@@ -245,6 +245,20 @@
 #define UNIX_AMD64_ABI_ONLY(x)
 #endif // defined(UNIX_AMD64_ABI)
 
+#if defined(DEBUG) && !defined(OSX_ARM64_ABI)
+#define DEBUG_NOT_OSX_ARM64_ABI
```

Comment:
IMO these would be better left as, e.g. `defined(DEBUG) && !defined(OSX_ARM64_ABI)` at each use. I don't find the implied conjunction clear enough, though I see that it might be difficult to deal with the #defines below. Perhaps you could define something like `TRACK_ARG_SLOTS` or `DEBUG_ARG_SLOTS` and have a comment here about why it is not used for the `OSX_ARM64_ABIT` case?

Reply:
I would say it is a temporary solution, all code under `DEBUG_NOT_OSX_ARM64_ABI` will go away once the work is done and we check that the new byte mechanism is correct and matches the old results. However, I have found it useful to use a new define because now I can easily change it:
1) define it in release for non apple arm64;
2) replace `assert`-s with `noway_assert`-s and make debugging easier;
etc.

Reply:
Yes, I agree that it's not a huge deal since it's temporary, but I appreciate you making the changes~